### PR TITLE
Add vehicle type to differentiate between BUS, TRAIN & AIRCRAFT

### LINKED ISSUE #161: Add info about the type of carrier
Add field to each leg indicating whether it's train, bus or plane

### PR DESCRIPTION
**Pull request by jaroslav-kubicek**
Closes #161

### CODE REVIEW

File: src/booking/dataloaders/__tests__/__datasets__/booking.detail.json [hunk 1]
```diff
@@ -51,11 +51,11 @@
       "name": "LOT Polish Airlines"
     },
     "vehicle": {
-      "model": "DHC-8-400 Dash 8Q",
-      "seats": 78,
-      "type": "DH4",
-      "meal": [],
-      "manufacturer": "De Havilland Canada"
+      "manufacturer": null,
```
Comment (michalsanger) [inline src/booking/dataloaders/__tests__/__datasets__/booking.detail.json]: REST API stoped providing these data?

File: src/flight/types/outputs/Leg.js [hunk 2]
```diff
@@ -58,5 +72,30 @@ export default new GraphQLObjectType({
...
+        { id, vehicleType, bookingId }: Leg,
+        args: Object,
+        { dataLoader }: GraphqlContextType,
+      ): Promise<?VehicleType> => {
+        if (vehicleType) {
+          return vehicleType;
+        }
+
+        if (bookingId) {
+          const booking = await dataLoader.booking.load(bookingId);
```
Comment (michalsanger) [inline src/flight/types/outputs/Leg.js]: When is this needed? There can be a leg without `vehicleType` but fetching it again via booking API will load the value?

File: src/booking/dataloaders/__tests__/__datasets__/booking.detail.json (hunk 1 above)
Comment (jaroslav-kubicek) [reply src/booking/dataloaders/__tests__/__datasets__/booking.detail.json]: yes, only the type is non-null now 🤷‍♂️

File: src/flight/types/outputs/Leg.js (hunk 2 above)
Comment (jaroslav-kubicek) [reply src/flight/types/outputs/Leg.js]: yep, it's necessary due to "allbookings" query where this information is missing

Comment (jaroslav-kubicek) [reply src/flight/types/outputs/Leg.js]: I rather added explanation into comment

**Pull request merged**